src/session_pool/session.js

```diff
@@ -210,37 +218,46 @@ export class Session {
         }
     }
 
-    /**
-     * Persists puppeteer cookies to session for reuse.
-     * @param puppeteerCookies - cookie from puppeteer `page.cookies` method.
-     * @param url - Loaded url from page function.
-     */
-    putPuppeteerCookies(puppeteerCookies, url) {
-        const cookies = puppeteerCookies.map(puppeteerCookie => this._transformPuppeteerCookie(puppeteerCookie));
-
-        this.setCookies(cookies, url);
-    }
-
     /**
      * Set cookies to session cookieJar.
-     * Cookies array should be compatible with tough-cookie.
+     * Cookies array should be [tough-cookie](https://github.com/salesforce/tough-cookie)
+     * or [puppeteer](https://pptr.dev/#?product=Puppeteer&version=v2.0.0&show=api-pagecookiesurls) cookie compatible.
      * @param cookies {Array<Cookie>}
      * @param url {String}
      */
     setCookies(cookies, url) {
+        const isPuppeteerCookies = !!cookies[0] && !!cookies[0].name;
+
         for (const cookie of cookies) {
-            this.cookieJar.setCookieSync(cookie, url, { ignoreError: false });
+            let cookieToSave;
+
+            if (isPuppeteerCookies) {
+                cookieToSave = this._puppeteerCookieToTough(cookie);
```

Comment:
Is this correct? Puppeteer cookies are stored as tough, but the type is still TOUGH_COOKIE_TYPE

Reply:
I will remove it completely.